### ISSUE TITLE
Zyxel P-661HNU-Fx issues

### ISSUE THREAD

**Issue by tboHammu** [danielschwierzeck/u-boot-lantiq#20]
Hi, I am working on a Zyxel P-661HNU-Fx OpenWrt support and currently I am trying to make U-boot work. I built it from your openwrt/v2014.07 branch and it does boot via UART, but I cannot seem to make it work when booted from SPI (this board has a MX25L128 SPI chip). Is the u-boot.ltq.lzo.sfspl correct image for this board and not the regular u-boot.bin file? Also the network switch does not seem to be working/recognized at all when U-boot is booted from UART, when chainloaded from the stock bootloader it at least showed some signs of life (before I wiped out the flash completely ...)
Do you have any pointers on what I should check?

**Comment by danielschwierzeck** [comment]
You need to flash u-boot.ltq.lzo.sfspl because only that image contains the SPL and the init/download instructions for the AR9 BootROM. I have a P-661HNU-F3 which was tested with the openwrt/v2014.07 branch. 

Regarding network this is likely due to the missing backport of commit 704f3acfcf55343043bbed01c5fb0a0094a68e8a which is required for gcc > 6.x. I forgot to push it but that's now fixed.

**Comment by tboHammu** [comment]
And the file should be flashed directly to the start of the spi i.e offset 0? I've tried both dd'ing from linux and sf write from u-boot, but neither does make it boot, it just stops after the bootROM. BTW with 20MHz spispeed, writing resulted in completely garbled flash. With 10MHz it seems to write OK, but still no boot.

Do you happen to have a working build that I could test?

**Comment by danielschwierzeck** [comment]
> And the file should be flashed directly to the start of the spi i.e offset 0?

Yes, but there is a macro "run update-uboot-sf" to simplify this.

I just tested the board and indeed it doesn't work. I thought I only have migrated working boards from openwrt/2013.10 to openwrt/2014.07 but P661 is not fully working in openwrt/2013.10 as well :( 

IIRC I gave up on that board due to lack of time and because I couldn't figure out the switch reset GPIO and the BootROM issues. Unfortuneately I don't have time atm to debug this further. But at least networking is supposed to work in UART boot. Maybe you can figure out the switch reset GPIO, it's obviously not connected to the HRST line.

**Comment by tboHammu** [comment]
> I just tested the board and indeed it doesn't work.

Do you mean it does not boot when flashed? Or the network issue?
Because I found out I found out that my backup is missing the first 384Kb, so I have currently nothing to flash to the router :( and UART-booting is really starting to annoy me.
Do you remember what sort of BootROM issues you were having?
